pkg/alertmanager/statefulset.go

```diff
@@ -230,71 +230,86 @@ func makeStatefulSetSpec(logger *slog.Logger, a *monitoringv1.Alertmanager, conf
 		return nil, fmt.Errorf("failed to parse alertmanager version: %w", err)
 	}
 
-	amArgs := []string{
-		fmt.Sprintf("--config.file=%s", path.Join(alertmanagerConfigOutDir, alertmanagerConfigEnvsubstFilename)),
-		fmt.Sprintf("--storage.path=%s", alertmanagerStorageDir),
-		fmt.Sprintf("--data.retention=%s", a.Spec.Retention),
+	amArgs := []monitoringv1.Argument{
+		{Name: "config.file", Value: path.Join(alertmanagerConfigOutDir, alertmanagerConfigEnvsubstFilename)},
+		{Name: "storage.path", Value: alertmanagerStorageDir},
+		{Name: "data.retention", Value: string(a.Spec.Retention)},
 	}
 
 	if *a.Spec.Replicas == 1 && !a.Spec.ForceEnableClusterMode {
-		amArgs = append(amArgs, "--cluster.listen-address=")
+		amArgs = append(amArgs, monitoringv1.Argument{Name: "cluster.listen-address", Value: ""})
 	} else {
-		amArgs = append(amArgs, "--cluster.listen-address=[$(POD_IP)]:9094")
+		amArgs = append(amArgs, monitoringv1.Argument{Name: "cluster.listen-address", Value: "[$(POD_IP)]:9094"})
 	}
 
 	if a.Spec.ListenLocal {
-		amArgs = append(amArgs, "--web.listen-address=127.0.0.1:9093")
+		amArgs = append(amArgs, monitoringv1.Argument{Name: "web.listen-address", Value: "127.0.0.1:9093"})
 	} else {
-		amArgs = append(amArgs, "--web.listen-address=:9093")
+		amArgs = append(amArgs, monitoringv1.Argument{Name: "web.listen-address", Value: ":9093"})
 	}
 
 	if a.Spec.ExternalURL != "" {
-		amArgs = append(amArgs, "--web.external-url="+a.Spec.ExternalURL)
+		amArgs = append(amArgs, monitoringv1.Argument{Name: "web.external-url", Value: a.Spec.ExternalURL})
 	}
 
 	if version.GTE(semver.MustParse("0.27.0")) && len(a.Spec.EnableFeatures) > 0 {
-		amArgs = append(amArgs, fmt.Sprintf("--enable-feature=%v", strings.Join(a.Spec.EnableFeatures[:], ",")))
+		amArgs = append(amArgs, monitoringv1.Argument{
+			Name:  "enable-feature",
+			Value: strings.Join(a.Spec.EnableFeatures[:], ","),
+		})
+	}
+
+	if version.GTE(semver.MustParse("0.28.0")) && len(a.Spec.AdditionalArgs) > 0 {
+		for _, arg := range a.Spec.AdditionalArgs {
+			if arg.Name == "auto-gomemlimit.ratio" {
+				amArgs = append(amArgs, monitoringv1.Argument{
+					Name:  arg.Name,
+					Value: fmt.Sprintf("%v", arg.Value),
+				})
+			}
+		}
```

Comment:
We don't need to do anything with the `auto-gomemlimit.ratio` CLI argument. The operator performs no check on additional args, it's used as-is and will break if the user messed it up.
```suggestion
```

Reply:
thanks, fixed